packages/framework-provider-azure/src/library/event-delete-adapter.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  BoosterConfig,
+  EventEnvelope,
+  EventDeleteParameters,
+  FilterFor,
+  SnapshotDeleteParameters,
+  UUID,
+  EventEnvelopeFromDatabase,
+  EntitySnapshotEnvelopeFromDatabase,
+} from '@boostercloud/framework-types'
+import { getLogger } from '@boostercloud/framework-common-helpers'
+import { CosmosClient } from '@azure/cosmos'
+import { replaceOrDeleteItem, search } from '../helpers/query-helper'
+
+export async function findDeletableEvent(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  parameters: EventDeleteParameters
+): Promise<Array<EventEnvelopeFromDatabase>> {
+  const logger = getLogger(config, 'events-delete-adapter#findDeletableEvent')
+  const stringifyParameters = JSON.stringify(parameters)
+  logger.debug(`Initiating a deletable event search for ${stringifyParameters}`)
+
+  const eventStore = config.resourceNames.eventsStore
+  const filter = buildDeleteEventFilter(parameters.entityTypeName, parameters.entityID, parameters.createdAt)
+  const result = (await search(cosmosDb, config, eventStore, filter)) as Array<EventEnvelopeFromDatabase>
+  logger.debug(`Finished deletable event search for ${stringifyParameters}`)
+  return result
+}
+
+export async function findDeletableSnapshot(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  parameters: SnapshotDeleteParameters
+): Promise<Array<EntitySnapshotEnvelopeFromDatabase>> {
+  const logger = getLogger(config, 'events-delete-adapter#findDeletableSnapshot')
+  const stringifyParameters = JSON.stringify(parameters)
+  logger.debug(`Initiating a deletable snapshot search for ${stringifyParameters}`)
+
+  const eventStore = config.resourceNames.eventsStore
+  const filter = buildDeleteEntityFilter(parameters.entityTypeName, parameters.entityID, parameters.createdAt)
+  const result = (await search(cosmosDb, config, eventStore, filter)) as Array<EntitySnapshotEnvelopeFromDatabase>
+  logger.debug(`Finished deletable snapshot search for ${stringifyParameters}`)
+  return result
+}
+
+export async function deleteEvent(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  events: Array<EventEnvelopeFromDatabase>
+): Promise<void> {
+  const logger = getLogger(config, 'events-delete-adapter#deleteEvent')
+  const stringifyParameters = JSON.stringify(events)
+  logger.debug(`Initiating an event delete for ${stringifyParameters}`)
+
+  const eventStore = config.resourceNames.eventsStore
+  if (!events || events.length === 0) {
+    logger.warn('Could not find events to delete')
+    return
+  }
+  for (const event of events) {
+    const newEvent = buildNewEvent(event)
+    const partitionKey = partitionKeyBuilder(event.entityTypeName, event.entityID, 'event')
+    await replaceOrDeleteItem(cosmosDb, eventStore, config, event.id, partitionKey, newEvent)
+  }
```

Comment:
I guess that the only way to delete entries from CosmoDB is one by one, isn't it?

Reply:
Correct. The Cosmos DB client for TypeScript does not provide a method for deleting items in bulk.